src/main/cpp/src/shuffle_split_detail.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/utilities/traits.hpp>
+
+namespace spark_rapids_jni {
+
+namespace detail {
+
+/**
+ * @brief Functor for computing size of data elements for a given cudf type.
+ *
+ * Note: columns types which themselves inherently have no data (strings, lists,
+ * structs) return 0.
+ */
+struct size_of_helper {
+  template <typename T>
+  constexpr std::enable_if_t<!cudf::is_fixed_width<T>() && !std::is_same_v<T, cudf::string_view>,
+                             size_t>
+  operator()() const
+  {
+    return 0;
+  }
+
+  template <typename T>
+  constexpr std::enable_if_t<!cudf::is_fixed_width<T>() && std::is_same_v<T, cudf::string_view>,
+                             size_t>
+  operator()() const
+  {
+    return sizeof(cudf::device_storage_type_t<int8_t>);
+  }
+
+  template <typename T>
+  constexpr std::enable_if_t<cudf::is_fixed_width<T>(), size_t> __device__
+  operator()() const noexcept
+  {
+    return sizeof(cudf::device_storage_type_t<T>);
+  }
+};
+
+/**
+ * @brief Header for each partition.
+ */
+struct partition_header {
+  uint32_t magic_number;
+  uint32_t row_index;  // row index in the source table that this partition started at
+  uint32_t num_rows;
+  uint32_t validity_size;
+  uint32_t offset_size;
+  uint32_t data_size;
```

Comment:
The last value in here is supposed to be the number of columns.
`uint32_t num_columns;` This feels redundant, seeing how we need the schema for this to be read back in. So should I remove it from the CPU code or should we add it here?

Reply:
@liurenjie1024 the numColummns field in the header looks like it should be something that we do not need. Is there a reason we are including it instead of relying on the schema to get it?